app/providers/docker_classic.go

```diff
@@ -103,3 +106,32 @@ func (provider *DockerClassicProvider) GetState(name string) (instance.State, er
 		return instance.UnrecoverableInstanceState(name, fmt.Sprintf("container status \"%s\" not handled", spec.State.Status), provider.desiredReplicas)
 	}
 }
+
+func (provider *DockerClassicProvider) NotifyInsanceStopped(ctx context.Context, instance chan string) {
+	msgs, errs := provider.Client.Events(ctx, types.EventsOptions{
+		Filters: filters.NewArgs(
+			filters.KeyValuePair{Key: "scope", Value: "local"},
+			filters.KeyValuePair{Key: "type", Value: "container"},
+			filters.KeyValuePair{Key: "event", Value: "die"},
+		),
+	})
+
+	go func() {
+		for {
+			select {
+			case msg := <-msgs:
+				// Send the container that has died to the channel
+				instance <- msg.From
+			case err := <-errs:
+				if errors.Is(err, io.EOF) {
+					log.Debug("provider event stream closed")
+					close(instance)
```

Comment:
This doesnt follow "The Channel Closing Principle"

One general principle of using Go channels is don't close a channel from the receiver side and don't close a channel if the channel has multiple concurrent senders. In other words, we should only close a channel in a sender goroutine if the sender is the only sender of the channel.

See https://go101.org/article/channel-closing.html#:~:text=The%20Channel%20Closing%20Principle,only%20sender%20of%20the%20channel.